foghorn_python/scripts/vcf_comp_funcs.py

```diff
@@ -0,0 +1,138 @@
+	elif GT in { "./.", ".|."}:
+		string = "".join(["Sample# ", str(sample_index), " has a both alleles missing at pos ->  ", str(fields[:4])])
+		mis.append(string)
+		sample_index = sample_index + 1
+		return sample_index, mis
+	else:
+		sample_index = sample_index + 1
+		return sample_index
+
+def DS_comp(DS, lines_holder, sample_index):
+	if float(DS) != 0:
+		string= "".join([str(sample_index), ":", str(DS)])
+		lines_holder.append(string)
+		sample_index = sample_index + 1
+		return lines_holder, sample_index
+	else:
+		sample_index = sample_index + 1
+		return sample_index
+
+#def mult_DS(DS, lines_holderDS, sample_indexDS):
+#	if
+
+def mult_al(GT, lines_holderMS, sample_indexMS, mis):
+	if GT in {"2|2", "2/2"}:
+		string= "".join([str(sample_indexMS), ":", "2"])
+		lines_holderMS.append(string)
+		return sample_indexMS, lines_holderMS
+	elif GT in {"0|2", "2|0", "2/0", "0/2"}: # need to clarify the odds of seeing 1|2. This will most probably be a sequencing error. So for the moment I am only including the referernce "0"
+	# This is one implementation the second is to just add extra indices 
+		string= "".join([str(sample_indexMS), ":", "1"])
+		lines_holderMS.append(string)
+		return sample_indexMS, lines_holderMS
+	elif GT in {".|.","./."}:
+                string = "".join(["Sample# ", str(sample_index), " has a both alleles missing at pos ->  ", str(fields[:5])])
+                mis.append(string)
+                sample_indexMS = sample_indexMS + 1
+                return sample_indexMS, mis
+	elif GT in {".|2", "2|.", "2/.", "./2", ".|1", "1|.", "1/.", "./1",".|0", "0|.", "0/.", "./0"}:
+		string = "".join(["Sample# ", str(sample_index), " has asingle allele missing at pos ->  ", str(fields[:5])])
+                mis.append(string)
+                sample_indexMS = sample_indexMS + 1
+                return sample_indexMS, mis
+	else:
+		return None
+
+
+def GT(line, out, mis_vars):
+        if not line.startswith("##") and not line.startswith("#"):  # Need to spend time making robust and reporting file format and missingness errors
+                fields = line.strip().split()
+		if "GT" not in fields[8]:
+			sys.exit("No genotype tag in format field")# raise error that there is no GT fields in 
+                sample_index = sample_indexMS = 0
+                lines_holderMS = [] # refresh every iteration need to check if this has any effect on the memory management of large datasets
+		lines_holder = []
+		mis = []
+		lines_holder.extend(fields[:4]) # append first few columns 
+		if "," in fields[4]:
+			lines_holderMS.extend(fields[:4])
+			lines_holder.append(str(str(fields[4]).split(",")[0])) #ALT1
+               		lines_holderMS.append(str(str(fields[4]).split(",")[1])) # ALT2
+               		for column in fields[9:]:
+   	        		GT = column.strip().split(":")[0]
+                      		GT_comp(GT, lines_holder, sample_index, mis, fields)
+                     		sample_index = sample_index + 1
```

Comment:
What is sample_index used for? I can't see where it is used. If it is  a legacy please clean up.

Reply:
Oops! Now I see what sample_index is used for. Could you not use the for loop here, instead that iterating over the genotype calls for each sample in (renamed) GT_comp function?